### PR TITLE
[WIP] Using route_path to generate url (fixes #103)

### LINKED ISSUE #103: pyramid.route_path to build internal route_path
We are building URIs by hand at multiple places in the code (e.g. parent id, default subrequests) and we should probably use ``request.route_path()`` for that.

http://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.route_path

### PR DESCRIPTION
**Pull request by vsham20**
Refs #103

### CODE REVIEW

File: kinto/views/buckets.py [hunk 1]
```diff
@@ -55,8 +55,9 @@ def on_buckets_deleted(event):
 
         # Delete records.
         for collection in deleted_collections:
-            parent_id = '/buckets/%s/collections/%s' % (bucket['id'],
-                                                        collection['id'])
+            parent_id = route_path('bucket-record',
```
Comment (leplatrem) [inline kinto/views/buckets.py]: collection-record ?

File: kinto/views/collections.py [hunk 2]
```diff
@@ -63,8 +63,9 @@ def on_collections_deleted(event):
 
     for change in event.impacted_records:
         collection = change['old']
-        parent_id = '/buckets/%s/collections/%s' % (event.payload['bucket_id'],
-                                                    collection['id'])
+        parent_id = route_path('collecection-record',
```
Comment (leplatrem) [inline kinto/views/collections.py]: typo

**Comment by vsham20** [comment]
@leplatrem Could you help me out here, i am not getting why test cases are failing?

**Comment by leplatrem** [comment]
What is the error that you have when you run the tests locally ?

**Comment by vsham20** [comment]
@leplatrem https://www.irccloud.com/pastebin/rWSVtOrd/

**Comment by leplatrem** [comment]
If you scroll further you should be able to find some details about the original error

**Comment by vsham20** [comment]
@leplatrem yes, i tried to find the error but i can't find where its going wrong

File: kinto/plugins/default_bucket/__init__.py [hunk 3]
```diff
@@ -4,15 +4,13 @@
 from pyramid import httpexceptions
 from pyramid.settings import asbool
 from pyramid.security import NO_PERMISSION_REQUIRED, Authenticated
-
```
Comment (Natim) [inline kinto/plugins/default_bucket/__init__.py]: Why did you removed this lines?

File: kinto/plugins/default_bucket/__init__.py [hunk 4]
```diff
@@ -4,15 +4,13 @@
...
 from cliquet.errors import raise_invalid
 from cliquet.utils import build_request, reapply_cors, hmac_digest
 from cliquet.storage import exceptions as storage_exceptions
-
 from kinto.authorization import RouteFactory
 from kinto.views.buckets import Bucket
 from kinto.views.collections import Collection
-
+from pyramid import route_path
```
Comment (Natim) [inline kinto/plugins/default_bucket/__init__.py]: Please put this in the pyramid importation block sorted alphabetically

Comment (Natim) [inline kinto/plugins/default_bucket/__init__.py]: Actually it seems that you cannot import ``route_path``, but that you should be using ``request.route_path``

File: kinto/plugins/default_bucket/__init__.py [hunk 5]
```diff
@@ -29,7 +28,9 @@ def create_bucket(request, bucket_id):
 
     # Fake context to instantiate a Bucket resource.
     context = RouteFactory(request)
-    context.get_permission_object_id = lambda r, i: '/buckets/%s' % bucket_id
+    context.get_permission_object_id = (lambda r,
+                                        i: Request.route_path('bucket-record',
+                                                              id=bucket_id))
```
Comment (leplatrem) [inline kinto/plugins/default_bucket/__init__.py]: `route_path()` is a method. It means it has to be called on an instance of the class. Not the class itself.

In this case:
```python
bucket_uri = request.route_path('bucket-record', id=bucket_id)
context.get_permission_object_id = lambda r, i: bucket_uri
```

**Comment by leplatrem** [comment]
@vsham20 can we help on this ?

**Comment by vsham20** [comment]
@leplatrem yes I made Change according to your previous comment but it's terminating with import error.
And currently I am not working on this issue as my semester exams are going on.Will continue after exams are over. Sorry :)

**Comment by Natim** [comment]
@vsham20 You could have a look at how @apratti did it here: https://github.com/apratti/kinto/commit/2a6ef036fa7dc4af013095aafc3b897183c39447

**Comment by apratti** [comment]
This was for a school project. Glad to help.

**Comment by Natim** [comment]
@apratti feel free to file a PR if you want to help :+1:

**Comment by leplatrem** [comment]
Closing for lack of activity..
Thanks everyone involved!
Feel free to reopen and revamp the work on this!

**Comment by Natim** [comment]
I will probably do the mandatory changes. I though @vsham20 was going to do it that's why I didn't so far.

**Comment by Natim** [comment]
Revamped in #702

**Pull request closed without merging**